tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Lambda/LambdaCommon.cs

```diff
@@ -55,9 +63,9 @@ internal static Scope SendStartInvocation(ILambdaExtensionRequest requestBuilder
         return CreatePlaceholderScope(tracer, headers);
     }
 
-    internal static void SendEndInvocation(ILambdaExtensionRequest requestBuilder, Scope scope, bool isError, string data)
+    internal static void SendEndInvocation(ILambdaExtensionRequest requestBuilder, Scope scope, object state, bool isError, string data)
```

Comment:
In general we should try to avoid passing `object` around. C# developers like the strong type safety that the language provides and treating anything as `object` disable that language feature.

To avoid that, we should cast the `state` to `CallTargetState` as early as possible and pass it around as a `CallTargetState`, not an `object`.